app/src/main/java/com/devshish/internship/presentation/ui/MainActivity.kt

```diff
@@ -1,23 +1,58 @@
 package com.devshish.internship.presentation.ui
 
+import android.Manifest
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.core.app.ActivityCompat
 import androidx.navigation.NavController
 import androidx.navigation.fragment.NavHostFragment
 import androidx.navigation.fragment.findNavController
 import androidx.navigation.ui.setupWithNavController
 import com.devshish.internship.R
 import com.devshish.internship.databinding.ActivityMainBinding
 import com.devshish.internship.presentation.ui.utils.viewBinding
+import timber.log.Timber
 
 class MainActivity : AppCompatActivity() {
 
     private val binding by viewBinding(ActivityMainBinding::inflate)
     private lateinit var navController: NavController
 
+    private val requestPermissionLauncher =
+        registerForActivityResult(
+            ActivityResultContracts.RequestPermission()
+        ) { isGranted: Boolean ->
+            if (isGranted) {
+                Timber.i("Granted")
+            } else {
+                Timber.i("Denied")
+            }
+        }
+
+    private fun requestPermissions() {
+        when {
+            ActivityCompat.shouldShowRequestPermissionRationale(
```

Comment:
`ActivityCompat.shouldShowRequestPermissionRationale` is needed when app provides some UI for user , describing why user needs this permission.

https://developer.android.com/training/permissions/requesting
<img width="400" alt="image" src="https://user-images.githubusercontent.com/23121846/131873906-fed32502-8d00-4df9-9ed7-76ac6dd084da.png">


In your case - `ActivityCompat.shouldShowRequestPermissionRationale`  is useless, so you can remove this and just ask permission